File: src/leetcode/python/easy/ValidWordSquare_422.py
class ValidWordSquare_422(object):
    def validWordSquare(self, words):
        """
        :type words: List[str]
        :rtype: bool
        """
        row = len(words)
        for i in range(row):
            for j in range(len(words[i])):
                try:
                    if words[i][j] != words[j][i]:
                        return False
                except:
                    return False
        return True

File: src/leetcode/java/ValidWordSquare_422.java
package leetcode.java;

import java.util.List;

public class ValidWordSquare_422 {
  public boolean validWordSquare(List<String> words) {
    int row = words.size();
    for (int i = 0; i < row; i++) {
      for (int j = 0; j < words.get(i).length(); j++) {
        try {
          if (words.get(i).charAt(j) != words.get(j).charAt(i)) return false;
        } catch (Exception e) {
          return false;
        }
      }
    }
    return true;
  }
}
